File: ios/ZillitLCW/Models/Revision.swift
import Foundation

struct Revision: Identifiable, Codable {
    let id: String
    let projectId: String
    let revisionNumber: Int
    let revisionColor: String
    let typeColor: String
    let description: String
    let changedBy: PerformedBy
    let createdAt: Int64

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case projectId, revisionNumber, revisionColor, typeColor, description, changedBy, createdAt
    }
}

File: android/app/src/main/java/com/zillit/lcw/data/model/Revision.kt
package com.zillit.lcw.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Revision(
    @SerialName("_id") val id: String,
    val projectId: String,
    val revisionNumber: Int,
    val revisionColor: String = "White",
    val typeColor: String = "",
    val description: String = "",
    val changedBy: PerformedBy = PerformedBy(),
    val createdAt: Long = 0L,
    val updatedAt: Long = 0L
)
